src/App.tsx

```diff
@@ -1,223 +1,180 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Todo } from './types/Todo';
+import { getTodos, addTodo, deleteTodoById } from './api/todos';
+import { FilterType } from './types/FilterType';
+import { ErrorNotification } from
+  './components/ErrorNotification/ErrorNotification';
+import { TodoItem } from './components/TodoItem/TodoItem';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterTodo, setFilterTodo] = useState(FilterType.all);
+  const [error, setError] = useState('');
+
+  const [newTitle, setNewTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [deletingTodosIds, setDeletingTodosIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
 
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => setError('Unable to load a todos'));
+    }
+  }, []);
+
+  const addNewTodo = useCallback((event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!newTitle) {
+      setError('Title can\'t be empty');
+
+      return;
+    }
+
+    const newTodoAdding = async () => {
+      setIsAdding(true);
+
+      if (user) {
```

Comment:
You can check the existence of user in advance, so there will be no need to wrap a large piece of code in if statement. It will improve readability.
```suggestion
      if (!user) {

        return;
      }

    const newTodoAdding = async () => {
      setIsAdding(true);
```